src/crimson/os/seastore/segment_cleaner.h

```diff
@@ -318,20 +328,26 @@ class SpaceTrackerI {
 
   virtual void dump_usage(segment_id_t) const = 0;
 
+  virtual double calc_utilization(segment_id_t segment) const = 0;
+
   virtual void reset() = 0;
 
   virtual ~SpaceTrackerI() = default;
 };
 using SpaceTrackerIRef = std::unique_ptr<SpaceTrackerI>;
 
 class SpaceTrackerSimple : public SpaceTrackerI {
+  struct segment_bytes_t {
+    int64_t live_bytes = 0;
+    uint64_t total_bytes = 0;
```

Comment:
Is this always 0? I don't see anywhere it is updated.

Reply:
It is initialized to be segment size

Reply:
Got it. Then:
1. the segment size can be moved out of `segment_bytes_t` as all segments have the same size;
2. calc_utilization is different between simple and detailed;

Reply:
This should be a mistake, I intended to calculate the usage by `live_extent_bytes/allocated_extent_bytes`, no matter simple or detailed. The reason was that the `age` in the benefit calculation equation is actually the age of extents, so the `util` in the benefit calculation equation should also be the utilization of extents, otherwise the meaning of `(1 - util) * age` would be relatively vague as `util` and `age` are not representing the same entities.

Reply:
I intend to think `live_extent_bytes/segment_size` is more correct as it also consider non-extent available space in the segment, for example, there are out-dated delta buffers that may be large.

Reply:
after a second thought, I move to think segment size should be more appropriate, will change `calc_utilization`

> the segment size can be moved out of segment_bytes_t as all segments have the same size;

segment_bytes_t needs to contain total_bytes as multiple devices, with different segment sizes, may be used.

Reply:
OK, this is across devices.